state/accountsDB.go

```diff
@@ -779,22 +780,30 @@ func (adb *AccountsDB) commit() ([]byte, error) {
 	log.Trace("accountsDB.Commit started")
 	adb.entries = make([]JournalEntry, 0)
 
-	oldHashes := make(common.ModifiedHashes)
-	newHashes := make(common.ModifiedHashes)
+	hc := hashesCollector.NewDisabledHashesCollector()
+	if adb.mainTrie.GetStorageManager().IsPruningEnabled() {
```

Comment:
make a separate function for this 3 lines -> and could we have without NEW at every single commit? like you could have a cleanup function here -> might be faster.

Reply:
for this if you could have a factory - which creates the good structure by default.